quinn-udp/src/unix.rs

```diff
@@ -196,7 +182,12 @@ fn init(io: SockRef<'_>) -> io::Result<()> {
             if rc == -1 {
                 return Err(io::Error::last_os_error());
             }
-        } else {
+        }
+    }
+
+    if !is_ipv4 {
+        #[cfg(any(target_os = "linux", target_os = "freebsd"))]
```

Comment:
Is there any reason to believe this is needed? `IPV6_RECVPKTINFO` is standardized.

Reply:
Yes, IPV6_RECVPKTINFO is standardized. 
But for other OS, eg. macOS, it will fail test echo_v6 because in quinn/src/tests.rs we do not distinguish between ipv4 and ipv6.
So I tried to add the ipv4 (IP_RECVDSTADDR) part to macOS, too. 
BTW, It is interesting that macOS supports both IP_RECVDSTADDR & IP_PKTINFO. 
It seems iOS will also work but I have no devices to test. ( I have MacBook Air & a Linux PVE server which runs a FreeBSD vm.)